### ISSUE TITLE
[Question] Analyzer for missing member in Equals Method

### ISSUE THREAD

**Issue by LiorBanai** [meziantou/Meziantou.Analyzer#676]
HI,
I was wondering if there is already rule for checking missing members in Equals/GetHashcode methods..

let say I have the following class

```csharp
 public record Bookmark
 {
     public required BookmarkType Type { get; init; }
     public required MessageSource Source { get; init; }
     public required Guid Id { get; init; }

     public virtual bool Equals(Bookmark? other)
     {
         if (ReferenceEquals(null, other))
         {
             return false;
         }

         if (ReferenceEquals(this, other))
         {
             return true;
         }

         return Type == other.Type && Source == other.Source && Id.Equals(other.Id);
     }

     public override int GetHashCode()
     {
         return HashCode.Combine((int)Type, (int)Source, Id);
     }
 }
```

if I add later on new property and missing adding it to the method.. 

Can this be detected?

**Comment by meziantou** [comment]
It could be detected by an analyzer. For instance, you can validate that the same properties are read from both methods. However, there are valid use-cases to have different `HashCode` and `Equals` method. Sometimes, you want a HashCode that is faster, so you can check only a subset of the properties. Don't forget that `GetHashCode() => 0` is a valid implementation.

Also, you can avoid discrepancies by using a source generator such as https://github.com/diegofrata/Generator.Equals

**Comment by LiorBanai** [comment]
got it. Thanks!
